app/src/main/java/org/astraea/app/common/Cache.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.astraea.app.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Cache<K, V> {
+
+  public interface CacheLoader<K, V> {
+    V load(K key);
+  }
+
+  public interface RemovalListener<K, V> {
+    void onRemove(Map.Entry<K, V> entry);
```

Comment:
Why using `Entry`? `(K k, V v)` is more straightforward , right?

Reply:
Indeed, many thanks